StantUiIosLibrary/Classes/Bars/SearchBar/DefaultSearchBarDelegateExtension.swift

```diff
@@ -8,19 +8,65 @@
 import UIKit
 
 extension DefaultSearchBar: UISearchBarDelegate {
+    public func searchBarShouldBeginEditing(_ searchBar: UISearchBar) -> Bool {
+        let cancelNotEmptySearch = didTapCancelButton && searchBar.text != ""
+        checkCancelButtonVisibility(conditional: cancelNotEmptySearch, searchBar: searchBar)
+        return true
+    }
+    
+    public func searchBarTextDidEndEditing(_ searchBar: UISearchBar) {
+        checkCancelButtonVisibility(conditional: didTapCancelButton, searchBar: searchBar)
+        didTapCancelButton = false
+    }
+    
     public func searchBar(_ searchBar: UISearchBar, textDidChange searchText: String) {
-        searchOnTableView = searchText
+        searchOnTableView  = searchText
+        isSearchEnable     = true
+        didTapCancelButton = false
+        
         if searchText.isEmpty {
-            searchBar.perform(#selector(self.resignFirstResponder), with: nil, afterDelay: 0.1)
+            searchBarCancelButtonClicked(searchBar)
+            return
         }
-        delegate?.updateTableViewWith(search: searchText)
+        
+        delegate?.updateTableViewWith(search: searchOnTableView)
+        
     }
     
     public func searchBarSearchButtonClicked(_ searchBar: UISearchBar) {
-        searchBar.resignFirstResponder()
+        searchOnTableView  = searchBar.text ?? ""
+        didTapCancelButton = false
         
+        enableCancelButtonAndHidekeyboardOf(searchBar: searchBar)
         delegate?.updateTableViewWith(search: searchOnTableView)
-        self.searchBar?.endEditing(true)
+    }
+
+    public func searchBarCancelButtonClicked(_ searchBar: UISearchBar) {
+        didTapCancelButton          = true
+        isSearchEnable              = false
+        searchBar.text              = ""
+        searchOnTableView           = ""
+        searchBar.showsCancelButton = isSearchEnable
+        
+        searchBar.perform(#selector(self.resignFirstResponder), with: nil, afterDelay: 0.1)
+        delegate?.updateTableViewWith(search: "")
+    }
+    
+    fileprivate func checkCancelButtonVisibility(conditional: Bool, searchBar: UISearchBar) {
```

Comment:
Are you using this variable isSearchEnable? Or you are just using to set to the showsCancelButton? If so, you could remove those conditions

Reply:
It is true. I did the adjustment. :)